spec/system/admin/orders_spec.rb

```diff
@@ -263,14 +263,33 @@
           end
 
           it "orders alphabetically by order number" do
-            find("a", text: 'STATE').click
+            find("a", text: 'STATE').click # sets ascending ordering
             expect(page).to have_content "Loading"
             expect(page).to have_content(/#{order5.number}.*#{order4.number}.*#{order3.number}.*#{order2.number}/m)
             find("a", text: 'STATE').click # sets descending ordering
             expect(page).to have_content "Loading"
             expect(page).to have_content(/#{order2.number}.*#{order3.number}.*#{order4.number}.*#{order5.number}/m)
           end
         end
+
+        context "orders by payment state" do
+          before do
+            Spree::Payment.where(order_id: order2.id).update(amount: 50.0)
```

Comment:
It's good to use `update!` in specs. We may introduce new validations one day and this may fail. If we don't check the result of `update` then it may take us a while to find out why the spec fails.

Reply:
Right - I've forgotten that validation tip again. Thanks.  

It's interesting though, while it works fine in cases such as:

`order2.update!(email: "jkl@jkl.com")`

In others like:

`Spree::LineItem.where(order_id: order2.id).update!(quantity: 5)`

it throws:

```
     Failure/Error: Spree::LineItem.where(order_id: order2.id).update!(quantity: 5)
     
     NoMethodError:
       undefined method `update!' for #<ActiveRecord::Relation []>
       Did you mean?  update
```

(took me a bit :-D but...) I think this is because there are several line items on each order, and `update!` does not seem to work on an array. So, for the purpose of the spec, it's totally fine to update only the first line item only :+1: 

While this is true for line items, it is then not true for:
  
`Spree::Payment.where(order_id: order2.id).update!(amount: 50.0)`

`order2` has only one payment, so I would not think it is needed to call `first` here. This is not a blocker - it's all greens - but I guess it's good to know, if anyone has an idea :thinking:

Reply:
Oh, that's another issue: `update` on a collection (ActiveRecord::Relation) is confusing because it updates each record individually and you can't tell if it failed. If you want to be efficient then you can use `update_all` or your do each record manually with `each` or `first` when there's only one.

Also, have a look at https://api.rubyonrails.org/v5.1/classes/ActiveRecord/Relation.html#method-i-update.